fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/UpdateNonRdfSourcePersisterTest.java

```diff
@@ -0,0 +1,190 @@
+import static org.fcrepo.persistence.common.ResourceHeaderSerializationUtils.deserializeHeaders;
+import static org.fcrepo.persistence.common.ResourceHeaderSerializationUtils.serializeHeaders;
+import static org.fcrepo.persistence.common.ResourceHeaderUtils.newResourceHeaders;
+import static org.fcrepo.persistence.common.ResourceHeaderUtils.touchCreationHeaders;
+import static org.fcrepo.persistence.common.ResourceHeaderUtils.touchModificationHeaders;
+import static org.fcrepo.persistence.ocfl.impl.OCFLPersistentStorageUtils.getInternalFedoraDirectory;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+import org.fcrepo.kernel.api.models.ResourceHeaders;
+import org.fcrepo.kernel.api.operations.CreateResourceOperation;
+import org.fcrepo.kernel.api.operations.NonRdfSourceOperation;
+import org.fcrepo.persistence.api.WriteOutcome;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+import org.fcrepo.persistence.ocfl.api.FedoraToOCFLObjectIndex;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSession;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSessionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * @author whikloj
+ * @since 6.0.0
+ */
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class UpdateNonRdfSourcePersisterTest {
+
+    @Mock
+    private NonRdfSourceOperation nonRdfSourceOperation;
+
+    @Mock
+    private OCFLObjectSession session;
+
+    @Mock
+    private FedoraOCFLMapping mapping;
+
+    @Mock
+    private FedoraToOCFLObjectIndex index;
+
+    @Mock
+    private OCFLObjectSessionFactory objectSessionFactory;
+
+    @Mock
+    private WriteOutcome writeOutcome;
+
+    @Captor
+    private ArgumentCaptor<InputStream> userContentCaptor;
+
+    @Captor
+    private ArgumentCaptor<InputStream> headersIsCaptor;
+
+    @Mock
+    private OCFLPersistentStorageSession psSession;
+
+    private static final String RESOURCE_ID = "info:fedora/parent/child";
+
+    private static final String ROOT_RESOURCE_ID = "info:fedora/parent";
+
+    private static final String USER_PRINCIPAL = "fedoraUser";
+
+    private static final String CONTENT_BODY = "this is some example content";
+
+    private static final Long LOCAL_CONTENT_SIZE = Long.valueOf(CONTENT_BODY.length());
+
+    private static final String EXTERNAL_URL = "http://example.com/file.txt";
```

Comment:
none of these external variables are being used. Perhaps an external binary test should be added?

Reply:
This is also making me realize that the persister probably needs to be able to handle the case of switching from an internal to an external binary, which would entail deleting the binary (or is it writing a 0 byte file? I forget at the moment). Switching from an external to internal probably already works. Not sure if you want to handle that in this PR or make a followup.

Reply:
I think we want this in a separate PR / JIRA.  @bbpennel  would you mind creating a jira for this one?

Reply:
https://jira.lyrasis.org/browse/FCREPO-3158